### PR TITLE
Debian integration

### PR DESCRIPTION
**Pull request by electrosenpai**
Adding Debian 8 (Jessie) to the ssg repo,  only few test for debian 8 but the structure is ready (make debian8 work and create ssg). Checklist is not big but test are useful with only 1 profil.

### CODE REVIEW

**Comment by iankko** [comment]
[ok to test]

**Comment by electrosenpai** [comment]
My bad on that , I change accidentally a File on Fedora that make it fail, should I remake a pull request with the fix ? (I just took the Fedora on master and merge it with mine)

**Comment by iankko** [comment]
@electrosenpai 
> My bad on that , I change accidentally a File on Fedora that make it fail, should I remake a pull request with the fix ? (I just took the Fedora on master and merge it with mine)

If I understood the issue correctly, you have taken some Fedora file from current SSG ```master``` and made it to fail? If that's the case it's enough to fix the concrete file to start passing again (e.g. by checking out the current version from ```master``` repository).

On your branch you would do:

```git checkout Fedora/input/modified_file.xml```

to retrieve the original (currently present) repo version.

HTH, Jan.

**Comment by mpreisler** [comment]
You have accidentally committed `config/oval.config`, please remove that change from the pull request.

**Comment by electrosenpai** [comment]
> You have accidentally committed config/oval.config, please remove that change from the pull request.

That's not the only file. I just totally fail and overwrite some file on Fedora directory, I'm working on it it will fix soon, sorry for this

**Comment by electrosenpai** [comment]
Ok I finally found what was wrong : 

we had in shared a oval definition "installed_OS_is_debian8" and the platform was wrong, it should be only on debian8 but it was on all platform ...

I fix it and I will push.

**Comment by electrosenpai** [comment]
I just retry the command that fails but it works but  :

```
$ make validate
/bin/sh: 1: Syntax error: Bad fd number
oscap xccdf validate-xml output/ssg-fedora-xccdf.xml
oscap oval validate-xml --schematron output/ssg-fedora-oval.xml
oscap ds sds-validate output/ssg-fedora-ds.xml
Skipping ShellCheck analysis, ensure shellcheck executable is present in the PATH!
cd output; ../utils/verify-references.py --rules-with-invalid-checks --ovaldefs-unused ssg-fedora-xccdf.xml
```

I have not idea what's wrong here , Fedora file is the same that master.

File: Debian/8/input/oval/idtranslate.py [hunk 1]
```diff
@@ -0,0 +1 @@
+../../transforms/idtranslate.py
```
Comment (redhatrises) [inline Debian/8/input/oval/idtranslate.py]: This softlink should be deleted.

**Comment by redhatrises** [comment]
@electrosenpai the following files should be deleted:

- `Debian/8/transforms/idtranslate.py`
- `Debian/8/transforms/idtranslate.pyc`
- `Debian/8/transforms/relabelids.py
- `Debian/8/transforms/combinechecks.py`
- `Debian/8/transforms/cpe_generate.py`
- `Debian/8/utils/verify-input-sanity.py`
- `Debian/8/input/oval/oval_5.11/.empty`

- `Debian/8/input/oval/testcheck.py` should match `Fedora/input/oval/testcheck.py` (Check the master branch)
- `Debian/8/utils/verify-references.py` should match `Fedora/utils/verify-references.py` (Check the master branch)

**Comment by electrosenpai** [comment]
> Debian/8/transforms/cpe_generate.py

I'm actually using this script in my Makefile

**Comment by redhatrises** [comment]
>> Debian/8/transforms/cpe_generate.py

>I'm actually using this script in my Makefile 

Your Makefile should use `$(SHARED)/$(TRANS)/cpe_generate.py` instead, and `Debian/8/transforms/cpe_generate.py` should be deleted.

**Comment by electrosenpai** [comment]
Ok , I can't push it right now but I will do it this night

**Comment by electrosenpai** [comment]
I'm thinking about checkout all file from master and adding Debian properly because It should not modify the Fedora file and it does not make sense

**Comment by iankko** [comment]
The issue is that forked repo is already outdated with the current master:
&nbsp; &nbsp; [1] https://github.com/electrosenpai/scap-security-guide/tree/master => 
``` This branch is 142 commits behind OpenSCAP:master. ``` => This PR has conflicts that need to be resolved first prior this can be merged (even if it could build).

Therefore:
* As the very <b>first thing</b> [synchronize your ```master``` branch with the current's SSG ```master```](https://help.github.com/articles/syncing-a-fork/),
* Once it's synced, create a completely new branch,
* Export the relevant changes from the current ```Debian_8``` branch via:
```
$ git clone -b Debian_8 https://github.com/electrosenpai/scap-security-guide.git
$ git format-patch HEAD~21
0001-initial-integration-of-Debian-8-in-ssg.-Not-yet-func.patch
0002-Adding-Debian-in-shared-combineovals.py.patch
0003-Fixing-and-adding-debian-in-shared-transforms-combin.patch
0004-Enhancement-DEBIAN-taking-old-Fedora-and-apply-chang.patch
0005-Enhancement-DEBIAN-taking-old-Fedora-and-apply-chang.patch
0006-Enhancement-DEBIAN-Test-correctly-working-debian-8-i.patch
0007-add-updated-version-of-Debian-8-support.-Not-yet-usi.patch
0008-updated-Debian-8-xccdf-checklist.-Add-first-support-.patch
0009-renamed-input-checks-into-input-oval.-Replaced-old-f.patch
0010-still-nearer-to-RHEL-Makefile.-Yet-not-completly-cle.patch
0011-corrected-ssg-URL.-Clean-font-matter-signature.patch
0012-add-more-complete-makefile-for-ocil-build.-Try-to-ad.patch
0013-deleting-DSA-check-from-xccdf-checklist-by-now.patch
0014-merge-standard-Makefile-into-proper-debian-8-build.-.patch
0015-add-oval_5.11-dir-for-5.11-OVAL-support.-No-5.11-spe.patch
0016-add-oval_5.11-dir-for-5.11-OVAL-support.-No-5.11-spe.patch
0017-add-oval_5.11-dir-for-5.11-OVAL-support.-No-5.11-spe.patch
0018-Fixing-shared-oval-installed_OS_is_debian8.xml.patch
0019-Debian-softlink-idtranslate.py-delete.patch
0020-DEBIAN-MAKEFILE-Adding-path-to-root-directory-for-sc.patch
0021-DEBIAN-Fixing-typo-in-updating.xml-rpm-apt.patch
```
* Then filter out the aforementioned changes to keep truly only those relevant to Debian, therefore:
 * 0001-initial-integration-of-Debian-8-in-ssg.-Not-yet-func.patch,
 * 0002-Adding-Debian-in-shared-combineovals.py.patch,
 * 0003-Fixing-and-adding-debian-in-shared-transforms-combin.patch,
 * 0004-Enhancement-DEBIAN-taking-old-Fedora-and-apply-chang.patch,
 * <b> Throw away 0005-Enhancement-DEBIAN-taking-old-Fedora-and-apply-chang.patch </b> (<b> since this one is doing unwanted changes in ```config/oval.config``` too </b>),
 * 0006-Enhancement-DEBIAN-Test-correctly-working-debian-8-i.patch,
 * 0007-add-updated-version-of-Debian-8-support.-Not-yet-usi.patch,
 * 0008-updated-Debian-8-xccdf-checklist.-Add-first-support-.patch,
 * 0009-renamed-input-checks-into-input-oval.-Replaced-old-f.patch,
 * 0010-still-nearer-to-RHEL-Makefile.-Yet-not-completly-cle.patch,
 * 0011-corrected-ssg-URL.-Clean-font-matter-signature.patch,
 * 0012-add-more-complete-makefile-for-ocil-build.-Try-to-ad.patch,
 * 0013-deleting-DSA-check-from-xccdf-checklist-by-now.patch,
 * 0014-merge-standard-Makefile-into-proper-debian-8-build.-.patch,
 * 0015-add-oval_5.11-dir-for-5.11-OVAL-support.-No-5.11-spe.patch,
 * 0016-add-oval_5.11-dir-for-5.11-OVAL-support.-No-5.11-spe.patch,
 * 0017-add-oval_5.11-dir-for-5.11-OVAL-support.-No-5.11-spe.patch,
 * 0018-Fixing-shared-oval-installed_OS_is_debian8.xml.patch,
 * 0019-Debian-softlink-idtranslate.py-delete.patch,
 * <b> Throw away all the Fedora changes from </b> 0020-DEBIAN-MAKEFILE-Adding-path-to-root-directory-for-sc.patch patch (<b> e.g. simply by deleting all the Fedora paths related patch hunks </b>),
  * 0021-DEBIAN-Fixing-typo-in-updating.xml-rpm-apt.patch

* Once these patches have been isolated (patch 0005-Enhancement-DEBIAN-taking-old-Fedora-and-apply-chang.patch <b> thrown away </b> and patch ```0020-DEBIAN-MAKEFILE-Adding-path-to-root-directory-for-sc.patch patch``` not to contain changes related to ```Fedora``` paths), perform the following,

* Based on the <b> synchronized fork </b> create new branch (I will call it ```Debian_8_next``` in the text below),
* Clone that ```Debian_8_next``` branch from the <b> synced </b> fork:
```
$ git clone -b Debian_8_next git@github.com:electrosenpai/scap-security-guide.git
```
* Ensure you are on the ```Debian_8_next``` branch:
```
$ git status
On branch Debian_8_next
Your branch is up-to-date with 'origin/Debian_8_next'.
nothing to commit, working directory clean
```
* Finally apply those patches from the above list (except <b>0005-</b> one, and with modified <b>0020-</b> one not to contain some Fedora paths changes) in <b>reverse</b> order like mentioned below: [*]
```
$ git am 0001-initial-integration-of-Debian-8-in-ssg.-Not-yet-func.patch
$ git am 0002-Adding-Debian-in-shared-combineovals.py.patch
...
$ git am 0021-DEBIAN-Fixing-typo-in-updating.xml-rpm-apt.patch
```

This way you should get a branch:
* that will be <b> synchronized </b> with the current SSG master (=> there won't be conflicts present),
* that will truly contain <b> only changes </b> relevant to ```Debian/8``` product introduction and <b> nothing else </b>

HTH, Jan.

---
[*] E.g. you would first apply patch <b>0001</b> on top of the synced branch, then <b>0002</b> etc. till <b>0021</b> excluding <b>0005</b> one, and having <b>0020</b> modified first.

**Comment by electrosenpai** [comment]
@iankko thank's for this tutorial, but I can't apply the 0001 because the Makefiles are too different.

However I remake a branch as you ask (name debian_8_next), that is clean and don't touch other files

**Comment by electrosenpai** [comment]
The branch is now clean @iankko I followed your tutorial and there is the branch right there : 

The branch is name : debian_8_integration [https://github.com/electrosenpai/scap-security-guide/tree/debian_8_integration]. It's totally clean without any modification of Fedora and config/config_oval. Shoud I close this pull request and start a new one with the branch debian_8_integration ?

**Comment by iankko** [comment]
@electrosenpai 
> Shoud I close this pull request and start a new one with the branch debian_8_integration ?
Yes,  please.

**Pull request closed without merging**